oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java

```diff
@@ -103,6 +104,20 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 				providerConfiguration, MediaType.APPLICATION_JSON, httpResponse);
 	}
 
+	/**
+	 * Sets the {@link HttpMessageConverter} used for converting the {@link OidcProviderConfiguration}
+	 * from and to HTTP requests and responses
+	 *
+	 * @param providerConfigurationHttpMessageConverter the {@link HttpMessageConverter} used for converting a
+	 * representation of the OpenID Provider Configuration from and to HTTP requests and responses
+	 *
+	 * @since 0.2.3
+	 */
+	public void setProviderConfigurationHttpMessageConverter(HttpMessageConverter<OidcProviderConfiguration> providerConfigurationHttpMessageConverter) {
```

Comment:
The thing about exposing `HttpMessageConverter<OidcProviderConfiguration>` is that it handles both read and write, but this `Filter` will only ever use the write. I wonder if we should just expose `Converter<OidcProviderConfiguration, Map<String, Object>>`, which ultimately hooks into `OidcProviderConfigurationHttpMessageConverter.setProviderConfigurationParametersConverter()`

Reply:
hi @jgrandja. I was thinking about this as well when I wrote the JavaDoc for the setter :-) 

I think it's a good idea to only expose the setter for the converter as the read method will never be used.